lib/enum/basic_enum.rb

```diff
@@ -78,73 +105,85 @@ def groups_of(num)
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-each_with_object
   # Tip: You'll have to do .each_with_object(Hash.new(0)) to create a hash to start with. Do not save the hash in a variable. You may use the incrementor method here.
   def element_frequency_count
+    collection.each_with_object(Hash.new(0)) { |key,value| value[key] += 1 }
   end
 
-
   # Refer to find_all / select
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-find-all
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-select
   def elements_ending_in_er
+    collection.map { |i| i.reverse}.find_all { |i| i[0...2] == "re" }.map{ |i| i.reverse}
```

Comment:
Interesting method here. It works, but it's possible to make it a lot shorter.

The first way would be to use negative numbers to go backwards from the end of the word. So `word[-1]` is the last character and `word[-2]` is the second-to-last character. Then you only need a single find_all method (or select, which is more common). 

Result: `collection.find_all { |word| word[-2..-1] == "er" }` or `collection.select { |word| word[-2..-1] == "er" }`

The other way would be to use Ruby's built-in end_with? method.